### ISSUE TITLE
no ambiguous antecedents (plugins documentation edition)

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5156]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5156|
|Type|enhancement|
|Created|2011-06-16 23:05:42Z|
|Branch|https://github.com/twisted/twisted/tree/ambiguous-antecedents-5156|

In the "Alternate Plugin Packages" section of the plugin howto document, these sentences appear:

    However, if you have developed an interface of your own, you may want to mandate that plugins for it are installed in your own plugins package, rather than in Twisted's. In this case, you probably also want to support yourproject/plugins/ directories for ease of development. To do so, you should make the !__init!__.py for that package contain at least the following lines.

Followed by a code snippet to put in said `__init__.py`.  However, the phrasing of the last sentence, _for that package_, is ambiguous.  It could refer to either `yourproject` or `yourproject.plugins`.  It should be clarified that this documentation is referring to the `__init__.py` for `yourproject.plugins`, ie `yourproject/plugins/__init__.py`.

Attachments:

* [plugins-ambiguous-antecedents-5156.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/214/214b42078b5891c09ee432a77af09d555b988b78/1b9c5b3ed419ab9dc8e09197e484d20de3d1590a.patch) (920 bytes) - added by indigo on 2011-10-04 15:22:23Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5156 5156
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_ambiguous_antecedents_5156 branches/ambiguous-antecedents-5156
branch_author__indigo indigo
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1308265542000000 1308265542000000
changetime__1317760529000000 1317760529000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jafd's avatar" src="https://avatars.githubusercontent.com/u/301073?s=50" width="50" height="50">](https://github.com/jafd)<a name="note_2"></a>|@jafd commented|
|-|-|

  In this case, you probably also want to support yourproject/plugins/ directories for ease of development. To do so, you need to make yourproject/plugins directory a regular Python package, and its !__init!__.py should contain at least the following lines.

Would that be okay?

**Comment by twisted-trac** [comment]
|[<img alt="bitglue's avatar" src="https://avatars.githubusercontent.com/u/1232472?s=50" width="50" height="50">](https://github.com/bitglue)<a name="note_3"></a>|@bitglue commented|
|-|-|

please review. merge with .misc topfile?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [32747]) Branching to 'ambiguous-antecedents-5156'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

(In [32748]) Apply plugins-ambiguous-antecedents-5156.patch

refs [#5156](https://github.com/twisted/twisted/issues/5156)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @exarkun|
|-|-|

Thanks for taking a look at this.

  1. There's a typo in the first new sentence.
  1. I don't think supporting these plugins directories is an example.  It's just something you might want to do.  So the "For example" is off-putting.

Otherwise fine.  I'll tweak that first sentence trivially and apply.  Thanks again.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

(In [32749]) Fix typo, tweak phrasing

refs [#5156](https://github.com/twisted/twisted/issues/5156)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [32751]) Merge ambiguous-antecedents-5156

Author: indigo
Reviewer: exarkun
Fixes: [#5156](https://github.com/twisted/twisted/issues/5156)

Clarify the language in the plugin howto regarding how to create new
splittable plugin-holding directories (in the way twisted/plugins/ can
exist in multiple places and have all contents merged).
